src/olympia/promoted/tests/test_models.py

```diff
@@ -427,6 +427,54 @@ def test_approve_for_addon(self):
         ).exists()
 
 
+class TestPromotedGroupManager(TestCase):
+    def setUp(self):
+        self.addon: Addon = addon_factory()
+        self.promoted_group = PromotedGroup.objects.get(
+            group_id=PROMOTED_GROUP_CHOICES.SPOTLIGHT
+        )
+        self.application_id = applications.FIREFOX.id
+
+    def test_approved_for(self):
+        # addon has a promotion, but no associated version (no approval)
+        promotion = PromotedAddonPromotion.objects.create(
+            addon=self.addon,
+            promoted_group=self.promoted_group,
+            application_id=self.application_id,
+        )
+        assert not PromotedGroup.objects.approved_for(self.addon)
+
+        # now approved, should appear
+        PromotedAddonVersion.objects.create(
+            version=self.addon.current_version,
+            promoted_group=self.promoted_group,
+            application_id=self.application_id,
+        )
+        assert self.promoted_group in PromotedGroup.objects.approved_for(self.addon)
+
+        # if the current version changes (the group was not
+        # carried over) the approval is no longer valid
+        self.addon._current_version = version_factory(addon=self.addon)
+        assert not PromotedGroup.objects.approved_for(self.addon)
+
+        # alternatively, addon has a non-pre-reviewed promoted group
+        strategic_group = PromotedGroup.objects.get(
+            group_id=PROMOTED_GROUP_CHOICES.STRATEGIC
+        )
+        promotion.promoted_group = strategic_group
+        promotion.save()
+        assert strategic_group in PromotedGroup.objects.approved_for(self.addon)
+
+    def test_all_for(self):
+        assert not PromotedGroup.objects.all_for(self.addon)
```

Comment:
To validate the assumption, add a query for the PromotedAddonPromotions for that addon, expecting 0 at first.